ndk/src/font.rs

```diff
@@ -0,0 +1,401 @@
+    pub fn weight(&self) -> FontWeight {
+        FontWeight(unsafe { ffi::AFont_getWeight(self.ptr.as_ptr()) })
+    }
+
+    /// Return true if the current font is italic, otherwise returns false.
+    pub fn is_italic(&self) -> bool {
+        unsafe { ffi::AFont_isItalic(self.ptr.as_ptr()) }
+    }
+}
+
+#[cfg(feature = "api-level-29")]
+impl Drop for Font {
+    fn drop(&mut self) {
+        unsafe { ffi::AFont_close(self.ptr.as_ptr()) }
+    }
+}
+
+/// Corresponds to [`AFAMILY_VARIANT_*`].
+///
+/// [`AFAMILY_VARIANT_*`]: https://developer.android.com/ndk/reference/group/font#group___font_1gga96a58e29e8dbf2b5bdeb775cba46556ea662aafc7016e35d6758da93416fc0833
+#[repr(u32)]
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum FamilyVariant {
+    /// A family variant value for the compact font family variant.
+    /// The compact font family has Latin-based vertical metrics.
+    Compact = ffi::AFAMILY_VARIANT_COMPACT as _,
+    /// A family variant value for the system default variant.
+    Default = ffi::AFAMILY_VARIANT_DEFAULT as _,
+    /// A family variant value for the elegant font family variant.
+    /// The elegant font family may have larger vertical metrics than Latin font.
+    Elegant = ffi::AFAMILY_VARIANT_ELEGANT as _,
+}
+
+/// A native [`AFontMatcher *`]
+///
+/// [`AFontMatcher *`]: https://developer.android.com/ndk/reference/group/font#afontmatcher_create
+#[cfg(feature = "api-level-29")]
+#[derive(Debug)]
+pub struct FontMatcher {
+    ptr: NonNull<ffi::AFontMatcher>,
+}
+
+#[cfg(feature = "api-level-29")]
+impl FontMatcher {
+    /// Create an `FontMatcher` from a pointer
+    ///
+    /// # Safety
+    /// By calling this function, you assert that the pointer is a valid pointer to a native
+    /// `AFontMatcher`.
+    pub unsafe fn from_ptr(ptr: NonNull<ffi::AFontMatcher>) -> Self {
+        Self { ptr }
+    }
+
+    /// Returns the pointer to the native `AFontMatcher`.
+    pub fn ptr(&self) -> NonNull<ffi::AFontMatcher> {
+        self.ptr
+    }
+
+    /// Select the best font from given parameters.
+    ///
+    /// Creates a new [`FontMatcher`] object.
+    pub fn new() -> Self {
+        NonNull::new(unsafe { ffi::AFontMatcher_create() })
+            .map(|p| unsafe { FontMatcher::from_ptr(p) })
+            .expect("AFontMatcher_create returned NULL")
+    }
+
+    /// Set family variant to matcher.
+    ///
+    /// If this function is not called, the matcher performs with [`FamilyVariant::Default`].
+    pub fn set_family_variant(&mut self, family_variant: FamilyVariant) {
+        unsafe { ffi::AFontMatcher_setFamilyVariant(self.ptr.as_ptr(), family_variant as u32) }
+    }
+
+    /// Set font locales to matcher.
+    ///
+    /// If this function is not called, the matcher performs with empty locale list.
+    ///
+    /// # Arguments
+    /// * `language_tags`: a null character terminated comma separated IETF BCP47 compliant language tags.
```

Comment:
No need for a single argument? Also, drop "null character terminated" blah, that is implicit because of the `CStr` type :)

Reply:
This is copied from the original documentation; since the IETF thing is also mentioned in `Font::locale`, I think leaving it alone is better. I'll remove the "null character terminated" part.